python/src/aac/lang/_active_context.py

```diff
@@ -0,0 +1,165 @@
+"""The Active Context maintains the highly-contextual AaC DSL."""
+import copy
+from attr import Factory, attrib, attrs, validators
+
+from aac.parser import ParsedDefinition
+from aac.lang.definition_helpers import (
+    get_definition_by_name,
+    get_definitions_by_root_key,
+    search_definition,
+    remove_list_type_indicator,
+)
+
+
+@attrs(slots=True, auto_attribs=True)
+class ActiveContext:
+    """
+    A management and utility class for the contextual AaC domain-specific language.
+
+    Because the AaC DSL is self-defining, self-validating, and highly extensible, the DSL
+    structure and rules are highly dependent on contextual definition sources such as active
+    plugins, 3rd party definitions (libraries), and user-defined definitions.
+
+    Attributes:
+        context_definitions: The list of all definitions currently in the ActiveContext
+    """
+
+    definitions: list[ParsedDefinition] = attrib(default=Factory(list), validator=validators.instance_of(list))
+
+    def add_definition_to_context(self, parsed_definition: ParsedDefinition):
+        """
+        Add the ParsedDefinition to the list of definitions in the ActiveContext.
+
+        Args:
+            parsed_definition: The ParsedDefinition to add to the context.
+        """
+        self.definitions.append(copy.deepcopy(parsed_definition))
+
+        if parsed_definition.is_extension():
+            self._apply_extension_to_context(parsed_definition)
+
+    def add_definitions_to_context(self, parsed_definitions: list[ParsedDefinition]):
+        """
+        Add the list of ParsedDefinitions to the list of definitions in the ActiveContext, any extensions are added last.
+
+        Args:
+            parsed_definitions: The list of ParsedDefinitions to add to the context.
+        """
+
+        extension_definitions = get_definitions_by_root_key("ext", parsed_definitions)
+
+        # Avoiding the use of sorted() since we already deepcopy each definition as we add it to the context.
+        for parsed_definition in parsed_definitions:
+            if parsed_definition not in extension_definitions:
+                self.add_definition_to_context(parsed_definition)
+
+        for extension_definitions in extension_definitions:
+            self.add_definition_to_context(extension_definitions)
+
+    def get_root_keys(self) -> list[str]:
+        """
+        Get the list of root keys as defined in the ActiveContext.
+
+        Returns a list of strings indicating the current root keys in the active context. These
+        keys may differ from those provided by the core spec since the ActiveContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+        See :py:func:`aac.spec.get_root_keys()` for the list of root keys provided by the unaltered core AaC DSL.
```

Comment:
Formatting keys? Are you referencing the docstring `:py:func:`aac.spec.get_root_keys()` section?

Reply:
Honestly I'm not sure why I called them "keys", but yes. I suggested using formatting markers (maybe?) and was told not to because it made the "docstrings less readable". Just curious if we are doubling back on that, now.

Reply:
Hmm. I don't see why that would be a big issue. The formatting here is in the long description, so it's not really impacting the high-value areas like summary, args, and return. 

Do you know where that feedback occurred in the other review? I'm curious if there is something about that particular instance that influenced that feedback.

Reply:
My brain aggressively dumps memories, so I can't speak to a previous position without referring to a written conversation or decision. 

VSCode handles the back-tick formatting correctly from what I saw. The documentation I see indicates that they are necessary for some of the formatting and advanced docstring functionality in Sphinx, so I'm happy to argue for allowing backticks to support better documentation.